### ISSUE TITLE
buildkite pipeline file limitation - error

### ISSUE THREAD

**Issue by chrislovecnm** [bazelbuild/rules_python#1209]
I added a new example into CI and got the following error:

```
The number of tasks in one config file is limited to 80!
```

This is because of https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/bazelci.py#L650

A few observations about our ci.

1. We have a single file, I am not a buildkite expert, but I wonder if we can break it up into multiple files.
2. We are testing across multiple Linux platforms: Debian and Ubuntu. I am wondering if we can drop one of those.
3. We are running tests for both bzlmod and WORKSPACE files (not sure we can do anything here).  Maybe combine those into a single run that happens in the same task?
4. We have separate runs for each folder under "tests".  Can we combine these at all?

The TLDR; is that we have a lot of tests in one file.  I love tests, but we need to see if we can break them into different files or prune the number of tests.  I assume that we are going to get more and more tests for both bzlmod and WORKSPACE files.

**Comment by rickeylev** [comment]
Looking at the [bazel ci docs](https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/README.md), I don't see anything that indicates how to combine multiple things into one.

The basic problem is our test matrix is big.
  * 4 OSes.
  * x2 for workspace vs bzlmod
  * x2 for bazel version 5.4 and 6.x

So that's about ~16 as our base. Since so much of our functionality is based on repository rules, we have to use separate WORKSPACE or MODULE configurations to test things, most of which are their own run.

The only thing I can think of is to turn some of them into bazel-within-bazel tests. Those are kind of a headache to debug when they fail. Maybe if we can figure out a good edit-run-debug story for those, it'll be more tenable. I'm thinking if one can do `bazel build //integration:test`, then cd into the output dir and run `bazel test ...`, that would go a long ways.

cc @meteorcloudy @fweikert Bazel CI owners

**Comment by meteorcloudy** [comment]
> 4 OSes.
> x2 for workspace vs bzlmod
> x2 for bazel version 5.4 and 6.x

That's 16 different configs, why are you exceeding the limitation of 80 jobs? Also maybe https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/README.md#matrix-testing can make the configuration a bit simpler?

**Comment by rickeylev** [comment]
Yeah, we have something like 8 or so tests that have their own workspace/module files and are their own run. Not everything is tested in all 16 combinations, but it's close. You can see that the py proto library test has 10 combinations, pip parse has 9, etc.
